### ISSUE TITLE
New Asset Type: Spreadsheet

### ISSUE THREAD

**Issue by KingDevyn** [YoYoGames/GameMaker-Bugs#7103]
### Is your feature request related to a problem?

I find look up tables incredibly useful for just about every new big mechanic/system I implement. Hard coding them is fine, but could be much more organized if they could be edited in a spreadsheet editor that could link anything from numbers to strings to variable names to macros to strings (or even just numbers and strings would be nice).

### Describe the solution you'd like

A new asset type "spreadsheet" that can be converted into an array/ds grid/csv/+ at any point during runtime. For 3D+ arrays having pages for the Z dimension and beyond could also be a great function.
The new asset type would come with a spreadsheet editor with at least some basic functionality you'd expect (understanding not everything on excel and the like would make sense for this use case.)

### Describe alternatives you've considered

I've considered making .csv files from excel and putting them in the included files, however, this opens up the possibility of the end user getting to look at data you'd prefer not to be part of the experience (i.e. if used for dialogue look up tables) and also opens it up to being edited (which has its own merits if you want players to be able to re-ballance stat sheets and such.) Another limitation of this execution is that you don't have access to macros and other assets to link inside GMS which would be very helpful.

**Comment by Alphish** [comment]
I guess I'd rather call it "Tables" rather than spreadsheets, since I imagine the purpose would be storing rows of data in an organised manner, rather than making formula calculations across input cells. Other than that, I like it. ^^

**Comment by attic-stuff** [comment]
agreed, a table editor that turns into an array of arrays would be so tight

**Comment by nesrocks** [comment]
Great idea, I struggle with organizing my data in gml code format and tab spacing, and no matter how neatly I organize it it is not easy to maintain. A single new big word entry messes up all the formatting. And that's only for 2D tables, 3D are a big confusion.

**Comment by DragoniteSpam** [comment]
having a visual editor would make people who like to make data-heavy games such as RPGs very happy, i imagine

**Comment by iampremo** [comment]
Already planned ;)

**Comment by nullbunny** [comment]
That's great news! Is the asset type going to work with structs, too?
